### ISSUE TITLE
Timeout when using timestamp in require (runs instantly without)

### ISSUE THREAD

**Issue by leonardoalt** [melonproject/oyente#139]
Hi there,

I'm running a few tests, and I've come across this behavior for the following contract:
```
pragma solidity ^0.4.11;

contract T {
  uint sum0;
  uint sum1;
  
  function t(bool b) {
    require(block.timestamp < 1000);
    
    if (b) {
      sum1 += msg.value;
    }
    else {
      sum0 += msg.value;
    }

  }
}
```
With the contract like this, it times out.
Without the ``require`` line, it runs instantly.
I haven't checked the constraints that are passed to z3, but I would assume the branch that leads to the if/else has one constraint on the timestamp.

Any high level thoughts on why this happens?

**Comment by leonardoalt** [comment]
Ok, I figured what the problem was.
This is the piece of the disasm of the second code that I've posted related to the ``require`` command:
```
000086: PUSH1 0x5e
000088: JUMPI
000089: PUSH1 0x00
000091: PUSH1 0x00
000093: Missing opcode 0xfd (REQUIRE!!)
000094: JUMPDEST
```
If I change ``require`` to ``assert`` in the same code, it becomes:
```
000086: PUSH1 0x5a
000088: JUMPI
000089: Missing opcode 0xfe (ASSERT!!)
000090: JUMPDEST
```
The two extra ``PUSH``s are what break the stack afterwards, and lead to the TypeError.

My really-not-good-fix just to finish running my tests was to pop twice if the INVALID is followed by ``0xfd``. Both examples worked then.
